tests/common/snappi/snappi_fixtures.py

```diff
@@ -517,3 +521,355 @@ def cvg_api(snappi_api_serv_ip,
     yield api
     if getattr(api, 'assistant', None) is not None:
         api.assistant.Session.remove()
+
+
+def snappi_dut_base_config(duthost_list,
+                           snappi_ports,
+                           snappi_api):
+    """
+    Geenrate snappi API config and port config information for the testbed
+    Args:
+        conn_graph_facts (pytest fixture)
+        fanout_graph_facts (pytest fixture)
+        duthosts (pytest fixture): list of DUTs
+        rand_one_dut_hostname (pytest fixture): DUT hostname
+        snappi_api(pytest fixture): Snappi API fixture
+    Returns:
+        - config (obj): Snappi API config of the testbed
+        - port_config_list (list): list of port configuration information
+    """
+
+    """ Generate L1 config """
+
+    config = snappi_api.config()
+    tgen_ports = [port['location'] for port in snappi_ports]
+
+    new_snappi_ports = [dict(list(sp.items()) + [('port_id', i)])
+                        for i, sp in enumerate(snappi_ports) if sp['location'] in tgen_ports]
```

Comment:
no tgen

Reply:
Yes we can do this change from what i looked at, it will impact more files than this PR. Can we please merge this PR in and I will make a subsequent PR for the tgen name to be replaced on other testcases. To keep track of it i have raised an issue ( https://github.com/sonic-net/sonic-mgmt/issues/8210 ) and assigned it to myself.